desktop/cmp/grid/GridModel.js

```diff
@@ -291,7 +281,7 @@ export class GridModel {
             pull(newCols, emptyFlex).push(emptyFlex);
         }
 
-        this.setColumns(newCols);
+        this.columns = newCols;
```

Comment:
Wondering if there's any value in consistently using the setter internally as well as for the public API, to consolidate the `@action` and any other logic / checks that might prove useful. We also set the cols at end of `setGroupBy()`.

Cost would be looping / rebuilding the array again, but don't see that as material given grids will only have so many cols.

@lbwexler do you have an opinion here? I do not feel strongly just wanted to ask the question.

Reply:
Good thought, but I would be inclined to leave it as is, especially since the setter is now doing warning/validation.  Would be very annoying if my grid kept re-warning me every time I resized a column!

Reply:
That was my thought as well, want to avoid duplicate validation warnings if possible. One thing to note is that any caller of `setColumns` (such as the col chooser) could cause duplicate warnings. Actually, if the user removed all flex columns from the grid in the chooser and isn't using the emptyFlexCol, it would always cause a warning with this change. Not sure what (if anything) we should do for that case..